containers/Dockerfile.buildenv

```diff
@@ -104,6 +104,17 @@ RUN apt-get update -y \
 #
 # Build with GCC7 when easily possible to maximize ABI compatibility.
 #
+
+# Install cmake.  We need version 3.18.2 so that fortran builds work
+# with ninja.
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.18.2/cmake-3.18.2.tar.gz \
+    && tar -xzf cmake-3.18.2.tar.gz \
+    && cd cmake-3.18.2 \
+    && ./bootstrap --prefix=/usr/local \
+    && make \
+    && make install \
+    && rm -rf cmake*
+
```

Comment:
I don't like the location of this CMake build code, because,
- it falls under the comment for "dependencies not available through apt", which isn't accurate here, and
- it falls between a comment about using gcc7 for ABI compatibility and the code where gcc7 is explicitly used

So I'd suggest moving the CMake build above the preceding comment?

Reply:
agreed, also happy for you to squash